src/MSBuild.Conversion.Project/WinUI3Analyzers.cs

```diff
@@ -0,0 +1,234 @@
+                Console.WriteLine($"Converting Document {count} of {newProject.DocumentIds.Count}");
+                foreach (var analyzer in analyzers)
+                {
+                    Console.WriteLine($"Running {analyzer.GetType().Name} on {document.FilePath}");
+
+                    var codeFixProvider = GetCodeFixer(analyzer);
+                    if (codeFixProvider == null) continue;
+                    //Get all instances of that analyzer in the document
+                    IEnumerable<Diagnostic> analyzerDiagnostics;
+                    analyzerDiagnostics = GetSortedDiagnosticsFromDocument(analyzer, document, newProject);
+                    var attempts = analyzerDiagnostics.Count();
+                    //apply the changes to the document use total initial diagnostics as max attempts
+                    for (int i = 0; i < attempts; ++i)
+                    {
+                        var actions = new List<CodeAction>();
+                        var context = new CodeFixContext(document, analyzerDiagnostics.First(), (a, d) => actions.Add(a), CancellationToken.None);
+                        codeFixProvider.RegisterCodeFixesAsync(context).Wait();
+                        if (!actions.Any())
+                        {
+                            break;
+                        }
+                        //Apply fix and store in new solution
+                        document = ApplyFix(document, actions.ElementAt(0));
+                        newProject = document.Project;
+                        //apply the new document to the solution
+                        analyzerDiagnostics = GetSortedDiagnosticsFromDocument(analyzer, document, newProject);
+                        //check if there are analyzer diagnostics left after the code fix
+                        if (!analyzerDiagnostics.Any())
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+            // Remove Metadata references to avoid errors
+            newProject = newProject
+                .RemoveMetadataReference(CorlibReference)
+                .RemoveMetadataReference(SystemCoreReference)
+                .RemoveMetadataReference(CSharpSymbolsReference)
+                .RemoveMetadataReference(CodeAnalysisReference)
+                .RemoveMetadataReference(WindowsXamlReference)
+                .RemoveMetadataReference(MicrosoftXamlReference)
+                .RemoveMetadataReference(INotifyReference);
+
+            // Actually apply the accumulated changes and save them to disk. At this point
+            // workspace.CurrentSolution is updated to point to the new solution.
+            if (workspace.TryApplyChanges(newProject.Solution))
+            {
+                Console.WriteLine("Solution updated.");
+            }
+            else
+            {
+                Console.WriteLine("Update failed!");
```

Comment:
can we provide a better message than this?

Reply:
Reporting exception as well now.